mmv1/products/edgecontainer/Cluster.yaml

```diff
@@ -0,0 +1,515 @@
+          Project names are formatted as
+          `projects/<project-number>`.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: "membership"
+        default_from_api: true
+        immutable: true
+        description: |
+          The name of the managed Hub Membership resource associated to this cluster.
+          Membership names are formatted as
+          `projects/<project-number>/locations/global/membership/<cluster-id>`.
+  - !ruby/object:Api::Type::NestedObject
+    name: "networking"
+    required: true
+    description: |
+      Fleet related configuration.
+      Fleets are a Google Cloud concept for logically organizing clusters,
+      letting you use and manage multi-cluster capabilities and apply
+      consistent policies across your systems.
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: "clusterIpv4CidrBlocks"
+        item_type: Api::Type::String
+        immutable: true
+        required: true
+        description: |
+          All pods in the cluster are assigned an RFC1918 IPv4 address from these
+          blocks. Only a single block is supported. This field cannot be changed
+          after creation.
+      - !ruby/object:Api::Type::Array
+        name: "servicesIpv4CidrBlocks"
+        item_type: Api::Type::String
+        immutable: true
+        required: true
+        description: |
+          All services in the cluster are assigned an RFC1918 IPv4 address from these
+          blocks. Only a single block is supported. This field cannot be changed
+          after creation.
+      - !ruby/object:Api::Type::Array
+        name: "clusterIpv6CidrBlocks"
+        item_type: Api::Type::String
+        immutable: true
+        description: |
+          If specified, dual stack mode is enabled and all pods in the cluster are
+          assigned an IPv6 address from these blocks alongside from an IPv4
+          address. Only a single block is supported. This field cannot be changed
+          after creation.
+      - !ruby/object:Api::Type::Array
+        name: "servicesIpv6CidrBlocks"
+        item_type: Api::Type::String
+        immutable: true
+        description: |
+          If specified, dual stack mode is enabled and all services in the cluster are
+          assigned an IPv6 address from these blocks alongside from an IPv4
+          address. Only a single block is supported. This field cannot be changed
+          after creation.
+      - !ruby/object:Api::Type::Enum
+        name: "networkType"
+        default_from_api: true
+        immutable: true
+        description: |
+          IP addressing type of this cluster i.e. SINGLESTACK_V4 vs DUALSTACK_V4_V6.
+        values:
+          - :NETWORK_TYPE_UNSPECIFIED
+          - :SINGLESTACK_V4
+          - :DUALSTACK_V4_V6
+  - !ruby/object:Api::Type::NestedObject
+    name: "authorization"
+    required: true
+    immutable: true
+    default_from_api: true
```

Comment:
I'm not sure what the intention is for this field, but I believe that if you want it to be provided by the server when it isn't set, you will need to add `default_from_api: true` to each nested field. If that's not what we want, I believe this line should be removed.

Reply:
Gotcha, adding the `default_from_api: true` to each nested field. 
The field `authorization` is to define the user credential for accessing the cluster and is required for create operation.
The server will not modify the content, so setting default_from_api to each field should be the correct approach.